### PR TITLE
Add stream support for body request and for output body response

### PR DESCRIPTION
**Pull request by MatMaul**
For usage just pass a `Stream<List<int>>` as a body for requests, or specify `Response<Stream<List<int>>>` as a return type in your generator to get a stream.

### CODE REVIEW

File: chopper/lib/src/base.dart [hunk 1]
```diff
@@ -168,6 +169,9 @@ class ChopperClient {
 
     _requestController.add(req);
     final stream = await httpClient.send(await req.toBaseRequest());
```
Comment (lejard-h) [inline chopper/lib/src/base.dart]: maybe rename `final stream` to something else to avoid `stream.stream`

File: chopper/lib/src/interceptor.dart [hunk 2]
```diff
@@ -115,9 +115,12 @@ class HttpLoggingInterceptor
...
 
     var bytes;
-    if (response.base.body != null && response.base.body.isNotEmpty) {
-      chopperLogger.info(response.base.body);
-      bytes = ' (${response.base.bodyBytes.length}-byte body)';
+    if (response.base is http.Response) {
+      final resp = response.base as http.Response;
+      if (resp.body != null && resp.body.isNotEmpty) {
+        chopperLogger.info(resp.body);
+        bytes = ' (${response.bodyBytes?.length}-byte body)';
```
Comment (lejard-h) [inline chopper/lib/src/interceptor.dart]: `response.bodyBytes` can be null ? if so we should log something else instead of `null-byte body`

Comment (MatMaul) [reply chopper/lib/src/interceptor.dart]: It can't because we test just before that `base` is a Response and not a StreamedResponse.

**Comment by MatMaul** [comment]
I added one test case for each use case.

**Comment by MatMaul** [comment]
Also perhaps we should put a breaking change warning in the release note because I changed the type of `base`, and `body` is not accessible through it now. People should use `bodyBytes` instead now.

**Comment by lejard-h** [comment]
Indeed that can be problematic.
we could implement a `bodyString` getter to the Chopper `Response` that reimplement the `body` getter from `Response` https://pub.dev/documentation/http/latest/http/Response/body.html and yes introduce a breaking change in CHANGELOG

**Comment by MatMaul** [comment]
Indeed, I'll push a change with bodyString.

**Comment by MatMaul** [comment]
Done !

**Comment by MatMaul** [comment]
I added the warning in the changelog and bump the release to 2.4.0.

**Comment by lejard-h** [comment]
Thanks it's perfect 👍

**Comment by MatMaul** [comment]
Great, and I saw you added multipart and it's already released :+1:

**Pull request merged**